### PR TITLE
chore: Add credo config

### PR DESCRIPTION
**Pull request by cmaddox5**
I was going to disable a rule and realized we didn't have a credo config file. Generated one using `mix credo gen.config` and disabled the `Credo.Check.Refactor.Nesting` check. This PR can also be a forum to disagree/suggest more checks to disable.

### CODE REVIEW

**Comment by digitalcora** [comment]
What do you think of using only `disabled` and `extra` to tweak the config from the defaults, instead of having a "pinned" copy of all the checks? I think I'd like it if we could automatically inherit changes to the defaults with new Credo versions — for example I'm looking at that _"Checks scheduled for next check update (opt-in for now)"_ section and thinking that if we do things the "pinned" way, we're opting out of these checks indefinitely, not just for now.

**Comment by cmaddox5** [comment]
Yeah that's a good idea. Didn't think about how the config kinda forces you into a specific set of defaults out of the box. Cleans up the file quite a bit too.

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      #
+      name: "default",
+      #
+      # These are the files included in the analysis:
+      files: %{
+        #
+        # You can give explicit globs or simply directories.
+        # In the latter case `**/*.{ex,exs}` will be used.
+        #
+        included: ["lib/"],
```
Comment (digitalcora) [inline .credo.exs:24]: Interesting that it doesn't include `test/` by default. Inter-testing.

**Review by digitalcora — APPROVED** [review]

File: .credo.exs (hunk 1 above)
Comment (cmaddox5) [reply .credo.exs]: Woops, it actually does. It also comes with a bunch of directories it thinks you might have. When deleting those, I accidentally removed `test/`. Added it back.

**Pull request merged**